apps/whale-api/src/module.api/consortium.controller.spec.ts

```diff
@@ -0,0 +1,308 @@
+    ])
+    expect(info.total).toStrictEqual(5)
+  })
+
+  it('should filter and limit transactions at the same time', async () => {
+    const info = await controller.getTransactionHistory({ search: accountAlice, limit: 2 })
+
+    expect(info.transactions.length).toStrictEqual(2)
+    expect(info.transactions).toStrictEqual([
+      { txId: txIdMatcher, type: 'Burn', member: 'alice', tokenAmounts: [{ token: 'dETH', amount: '-1.00000000' }], address: accountAlice, block: 119 },
+      { txId: txIdMatcher, type: 'Mint', member: 'alice', tokenAmounts: [{ token: 'dETH', amount: '2.00000000' }], address: accountAlice, block: 114 }
+    ])
+    expect(info.total).toStrictEqual(3)
+  })
+
+  it('should get transactions upto a specific block height with a limit', async () => {
+    const info = await controller.getTransactionHistory({ limit: 3, maxBlockHeight: 124 })
+
+    expect(info.transactions.length).toStrictEqual(3)
+    expect(info.transactions).toStrictEqual([
+      { txId: txIdMatcher, type: 'Mint', member: 'bob', tokenAmounts: [{ token: 'dBTC', amount: '4.00000000' }], address: accountBob, block: 124 },
+      { txId: txIdMatcher, type: 'Burn', member: 'alice', tokenAmounts: [{ token: 'dETH', amount: '-1.00000000' }], address: accountAlice, block: 119 },
+      { txId: txIdMatcher, type: 'Mint', member: 'alice', tokenAmounts: [{ token: 'dETH', amount: '2.00000000' }], address: accountAlice, block: 114 }
+    ])
+    expect(info.total).toStrictEqual(5)
+  })
+
+  it('should return empty list of transactions for invalid search term', async () => {
+    const info = await controller.getTransactionHistory({ search: 'invalid-term', limit: 20 })
+
+    expect(info.transactions.length).toStrictEqual(0)
+    expect(info.total).toStrictEqual(0)
+  })
+
+  it('should not return other transactions from consortium members apart from mints or burns', async () => {
+    const { txid } = await alice.container.fundAddress(accountBob, 11.5)
+
+    const info = await controller.getTransactionHistory({ search: txid, limit: 20 })
+
+    expect(info.transactions.length).toStrictEqual(0)
+    expect(info.total).toStrictEqual(0)
+  })
+
+  it('should paginate properly', async () => {
+    const page1 = await controller.getTransactionHistory({ limit: 2 })
+
+    expect(page1).toStrictEqual({
+      transactions: [
+        {
+          type: 'Burn',
+          member: 'bob',
+          tokenAmounts: [{ token: 'dBTC', amount: '-2.00000000' }],
+          txId: txIdMatcher,
+          address: accountBob,
+          block: 129
+        },
+        {
+          type: 'Mint',
+          member: 'bob',
+          tokenAmounts: [{ token: 'dBTC', amount: '4.00000000' }],
+          txId: txIdMatcher,
+          address: accountBob,
+          block: 124
+        }
+      ],
+      total: 5
+    })
+
+    const page2 = await controller.getTransactionHistory({ limit: 2, maxBlockHeight: page1.transactions[page1.transactions.length - 1].block - 1 })
```

Comment:
Curious what would happen to pagination if there are 100 burnTokens at the same block (lez say block 124). Is this possible? 🤔 
Page1: first 20, (last block: 124)
Page2: next 20 (last block: 124)
Page3: next 20 and so on

If the above scenario is possible, would it still return the correct paginated results if the queries are the same?
```
// maxBlockHeight: lastBlock - 1
const page2 = await controller.getTransactionHistory({ limit: 20, maxBlockHeight: 123})
const page3 = await controller.getTransactionHistory({ limit: 20, maxBlockHeight: 123})
```

Reply:
Oh right, I don't have tests for transactions in the same block... I would be an issue. It is possible that consortium members do transactions in the same block although it might be rare but possible. 

If thats the case we can't paginate at all cause we can't cursor the tx list with transaction ids. 🤔

Reply:
For awareness, we'll be waiting for `ain` to provider a proper way to develop pagination for tx history CC: @thedoublejay @helloscoopa

Reply:
This is fixed on [`1d72f28`](https://github.com/JellyfishSDK/jellyfish/pull/1868/commits/1d72f28c75267c8c62012230f0eccc4bb002ba4f)